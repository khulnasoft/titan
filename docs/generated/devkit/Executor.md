# Type alias: Executor\<T\>

Ƭ **Executor**\<`T`\>: (`options`: `T`, `context`: [`ExecutorContext`](../../devkit/documents/ExecutorContext)) => `Promise`\<\{ `success`: `boolean` }\> \| `AsyncIterableIterator`\<\{ `success`: `boolean` }\>

Implementation of a target of a project

#### Type parameters

| Name | Type  |
| :--- | :---- |
| `T`  | `any` |

#### Type declaration

▸ (`options`, `context`): `Promise`\<\{ `success`: `boolean` }\> \| `AsyncIterableIterator`\<\{ `success`: `boolean` }\>

##### Parameters

| Name      | Type                                                        |
| :-------- | :---------------------------------------------------------- |
| `options` | `T`                                                         |
| `context` | [`ExecutorContext`](../../devkit/documents/ExecutorContext) |

##### Returns

`Promise`\<\{ `success`: `boolean` }\> \| `AsyncIterableIterator`\<\{ `success`: `boolean` }\>
