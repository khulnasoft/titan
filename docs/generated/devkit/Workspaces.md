# Class: Workspaces

**`Deprecated`**

This will be removed in v19. Use [readProjectsConfigurationFromProjectGraph](../../devkit/documents/readProjectsConfigurationFromProjectGraph) instead.

## Table of contents

### Constructors

- [constructor](../../devkit/documents/Workspaces#constructor)

### Methods

- [readWorkspaceConfiguration](../../devkit/documents/Workspaces#readworkspaceconfiguration)

## Constructors

### constructor

• **new Workspaces**(`root`): [`Workspaces`](../../devkit/documents/Workspaces)

#### Parameters

| Name   | Type     |
| :----- | :------- |
| `root` | `string` |

#### Returns

[`Workspaces`](../../devkit/documents/Workspaces)

## Methods

### readWorkspaceConfiguration

▸ **readWorkspaceConfiguration**(): [`ProjectsConfigurations`](../../devkit/documents/ProjectsConfigurations) & [`NxJsonConfiguration`](../../devkit/documents/NxJsonConfiguration)\<`string`[] \| `"*"`\>

#### Returns

[`ProjectsConfigurations`](../../devkit/documents/ProjectsConfigurations) & [`NxJsonConfiguration`](../../devkit/documents/NxJsonConfiguration)\<`string`[] \| `"*"`\>

**`Deprecated`**

Use [readProjectsConfigurationFromProjectGraph](../../devkit/documents/readProjectsConfigurationFromProjectGraph) instead.
