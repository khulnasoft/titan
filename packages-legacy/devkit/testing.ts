export * from '@titan/devkit/testing';
