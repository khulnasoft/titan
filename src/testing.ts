export {cold, hot} from 'jasmine-marbles';
export {readAll} from './utils/testing';
