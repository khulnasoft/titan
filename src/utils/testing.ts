import {Observable} from 'rxjs/Observable';
import {toArray} from 'rxjs/operator/toArray';
import {toPromise} from 'rxjs/operator/toPromise';

export function readAll<T>(o: Observable<T>): Promise<T[]> {
  return toPromise.call(toArray.call(o));
}
