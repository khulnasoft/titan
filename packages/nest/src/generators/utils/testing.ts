import { addProjectConfiguration, Tree } from '@titan/devkit';
import { createTreeWithEmptyWorkspace } from '@titan/devkit/testing';

export function createTreeWithNestApplication(appName: string): Tree {
  const tree = createTreeWithEmptyWorkspace();
  addProjectConfiguration(tree, appName, {
    root: `${appName}`,
    sourceRoot: `${appName}/src`,
    projectType: 'application',
    targets: {},
  });
  return tree;
}
