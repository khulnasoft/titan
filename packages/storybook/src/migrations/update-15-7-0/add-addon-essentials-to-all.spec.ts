import {
  addProjectConfiguration,
  ProjectConfiguration,
  Tree,
} from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import * as variousProjects from './test-configs/various-configs.json';
import addAddonEssentialsToAll from './add-addon-essentials-to-all';

describe('Add addon-essentials to project-level main.js files', () => {
  let tree: Tree;

  beforeEach(async () => {
    tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    addAllProjectsToWorkspace(tree);
  });

  it(`should add addon-essentials to main.js|ts where possible`, async () => {
    tree.write(
      '.storybook/main.js',
      `module.exports = {
        stories: [],
        addons: ['@storybook/addon-essentials', '@storybook/addon-knobs'],
      };
      `
    );
    await addAddonEssentialsToAll(tree);
    expect(
      tree.read('apps/main-vite/.storybook/main.js', 'utf-8')
    ).toMatchSnapshot();
    expect(
      tree.read('apps/main-vite-ts/.storybook/main.ts', 'utf-8')
    ).toMatchSnapshot();
    expect(
      tree.read('apps/main-webpack/.storybook/main.js', 'utf-8')
    ).toMatchSnapshot();
    expect(
      tree.read('libs/react-rollup/.storybook/main.js', 'utf-8')
    ).toMatchSnapshot();
    expect(
      tree.read('libs/react-rollup-3/.storybook/main.js', 'utf-8')
    ).toMatchSnapshot();
    expect(
      tree.read('libs/react-rollup-4/.storybook/main.js', 'utf-8')
    ).toMatchSnapshot();
    expect(
      tree.read('libs/react-vite/.storybook/main.js', 'utf-8')
    ).toMatchSnapshot();
    expect(
      tree.read('libs/react-vite-2/.storybook/main.js', 'utf-8')
    ).toMatchSnapshot();
  });

  it('should remove addon-essentials from root config file', async () => {
    tree.write(
      '.storybook/main.js',
      `module.exports = {
        stories: [],
        addons: ['@storybook/addon-essentials', '@storybook/addon-knobs'],
      };
      `
    );
    await addAddonEssentialsToAll(tree);
    expect(tree.read('.storybook/main.js', 'utf-8')).toMatchSnapshot();
  });

  it('should remove addon-essentials from root config typescript file', async () => {
    tree.write(
      '.storybook/main.ts',
      `
      import type { StorybookConfig } from '@storybook/core-common';

      export const rootMain: StorybookConfig = {
        addons: ['@storybook/addon-knobs', '@storybook/addon-essentials'],
      };
      `
    );
    await addAddonEssentialsToAll(tree);
    expect(tree.read('.storybook/main.ts', 'utf-8')).toMatchSnapshot();
  });

  it('should delete the root config file if it only has addon-essentials', async () => {
    tree.write(
      '.storybook/main.js',
      `module.exports = {
        stories: [],
        addons: ['@storybook/addon-essentials'],
      };
      `
    );
    await addAddonEssentialsToAll(tree);
    expect(tree.exists('.storybook/main.js')).toBeFalsy();
  });

  it(`should remove the root config import and delete root file`, async () => {
    tree.write(
      '.storybook/main.js',
      `module.exports = {
        stories: [],
        addons: ['@storybook/addon-essentials'],
      };
      `
    );

    await addAddonEssentialsToAll(tree);
    expect(
      tree.read('apps/main-vite/.storybook/main.js', 'utf-8')
    ).toMatchSnapshot();
    expect(
      tree.read('apps/main-vite-ts/.storybook/main.ts', 'utf-8')
    ).toMatchSnapshot();
    expect(
      tree.read('apps/main-webpack/.storybook/main.js', 'utf-8')
    ).toMatchSnapshot();
    expect(
      tree.read('libs/react-rollup/.storybook/main.js', 'utf-8')
    ).toMatchSnapshot();
    expect(
      tree.read('libs/react-rollup-3/.storybook/main.js', 'utf-8')
    ).toMatchSnapshot();
    expect(
      tree.read('libs/react-rollup-4/.storybook/main.js', 'utf-8')
    ).toMatchSnapshot();
    expect(
      tree.read('libs/react-vite/.storybook/main.js', 'utf-8')
    ).toMatchSnapshot();
    expect(
      tree.read('libs/react-vite-2/.storybook/main.js', 'utf-8')
    ).toMatchSnapshot();

    expect(tree.exists('.storybook/main.js')).toBeFalsy();
  });
});

export function addAllProjectsToWorkspace(tree: Tree) {
  for (const [name, project] of Object.entries(variousProjects)) {
    addProjectConfiguration(tree, name, project as ProjectConfiguration);
    if (project.targets['build-storybook']?.options?.configDir) {
      if (name === 'main-vite-ts') {
        tree.write(
          `${project.targets['build-storybook']?.options?.configDir}/main.ts`,
          `
          import { rootMain } from '../../../.storybook/main';
          import type { StorybookConfig } from '@storybook/core-common';
          import { mergeConfig } from 'vite';
          import viteTsConfigPaths from 'vite-tsconfig-paths';

          const config: StorybookConfig = {
            ...rootMain,
            core: { ...rootMain.core, builder: '@storybook/builder-vite' },
            stories: [
              ...rootMain.stories,
              '../src/app/**/*.stories.mdx',
              '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
            ],
            addons: [...(rootMain.addons || [])],
            async viteFinal(config: any) {
              return mergeConfig(config, {
                plugins: [
                  viteTsConfigPaths({
                    root: '../../../',
                  }),
                ],
              });
            },
          };

          module.exports = config;
          `
        );
      } else if (name === 'main-vite') {
        tree.write(
          `${project.targets['build-storybook']?.options?.configDir}/main.js`,
          `
          const rootMain = require('../../../.storybook/main');
          const { mergeConfig } = require('vite');
          const viteTsConfigPaths = require('vite-tsconfig-paths').default;

          module.exports = {
            ...rootMain,
            core: { ...rootMain.core, builder: 'webpack5' },
            stories: [
              ...rootMain.stories,
              '../src/app/**/*.stories.mdx',
              '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
            ],
            async viteFinal(config, { configType }) {
              return mergeConfig(config, {
                plugins: [
                  viteTsConfigPaths({
                    root: '../../../',
                  }),
                ],
              });
            },
          };
          `
        );
      } else if (name === 'react-vite-2') {
        tree.write(
          `${project.targets['build-storybook']?.options?.configDir}/main.js`,
          `
        some invalid stuff
        `
        );
      } else if (name === 'react-rollup') {
        tree.write(
          `${project.targets['build-storybook']?.options?.configDir}/main.js`,
          `
          // project: ${name}
          const rootMain = require('../../../.storybook/main');
          // Use the following syntax to add addons!
          rootMain.addons.push('@storybook/addon-a11y');
          rootMain.stories.push(
            ...['../src/lib/**/*.stories.mdx', '../src/lib/**/*.stories.@(js|jsx|ts|tsx)']
          );

          module.exports = rootMain;

          module.exports.core = { ...module.exports.core, builder: 'webpack5' };
          `
        );
      } else if (name === 'react-rollup-3') {
        tree.write(
          `${project.targets['build-storybook']?.options?.configDir}/main.js`,
          `
          // project: ${name}
          const rootMain = require('../../../.storybook/main');
          // Use the following syntax to add addons!
          rootMain.addons.push();
          rootMain.stories.push(
            ...['../src/lib/**/*.stories.mdx', '../src/lib/**/*.stories.@(js|jsx|ts|tsx)']
          );

          module.exports = rootMain;

          module.exports.core = { ...module.exports.core, builder: 'webpack5' };
          `
        );
      } else if (name === 'react-rollup-4') {
        tree.write(
          `${project.targets['build-storybook']?.options?.configDir}/main.js`,
          `
          // project: ${name}
          const rootMain = require('../../../.storybook/main');

          rootMain.stories.push(
            ...['../src/lib/**/*.stories.mdx', '../src/lib/**/*.stories.@(js|jsx|ts|tsx)']
          );

          module.exports = rootMain;

          module.exports.core = { ...module.exports.core, builder: 'webpack5' };
          `
        );
      } else {
        tree.write(
          `${project.targets['build-storybook']?.options?.configDir}/main.js`,
          `
          const rootMain = require('../../../.storybook/main');
          const { mergeConfig } = require('vite');
          const viteTsConfigPaths = require('vite-tsconfig-paths').default;

          module.exports = {
            ...rootMain,
            core: { ...rootMain.core, builder: 'webpack5' },
            stories: [
              ...rootMain.stories,
              '../src/app/**/*.stories.mdx',
              '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
            ],
            addons: [...rootMain.addons, '@nrwl/react/plugins/storybook'],
            webpackFinal: async (config, { configType }) => {
              // apply any global webpack configs that might have been specified in .storybook/main.js
              if (rootMain.webpackFinal) {
                config = await rootMain.webpackFinal(config, { configType });
              }

              // add your own webpack tweaks if needed

              return config;
            },
            async viteFinal(config, { configType }) {
              return mergeConfig(config, {
                plugins: [
                  viteTsConfigPaths({
                    root: '../../../',
                  }),
                ],
              });
            },
          };
          `
        );
      }
    }
  }
}
