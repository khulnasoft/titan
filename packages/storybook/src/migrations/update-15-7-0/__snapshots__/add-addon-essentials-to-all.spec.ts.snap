// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add addon-essentials to project-level main.js files should add addon-essentials to main.js|ts where possible 1`] = `
"const rootMain = require('../../../.storybook/main');
const { mergeConfig } = require('vite');
const viteTsConfigPaths = require('vite-tsconfig-paths').default;

module.exports = {
  ...rootMain,
  core: { ...rootMain.core, builder: 'webpack5' },
  stories: [
    ...rootMain.stories,
    '../src/app/**/*.stories.mdx',
    '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
  ],
  addons: ['@storybook/addon-essentials'],
  async viteFinal(config, { configType }) {
    return mergeConfig(config, {
      plugins: [
        viteTsConfigPaths({
          root: '../../../',
        }),
      ],
    });
  },
};
"
`;

exports[`Add addon-essentials to project-level main.js files should add addon-essentials to main.js|ts where possible 2`] = `
"import { rootMain } from '../../../.storybook/main';
import type { StorybookConfig } from '@storybook/core-common';
import { mergeConfig } from 'vite';
import viteTsConfigPaths from 'vite-tsconfig-paths';

const config: StorybookConfig = {
  ...rootMain,
  core: { ...rootMain.core, builder: '@storybook/builder-vite' },
  stories: [
    ...rootMain.stories,
    '../src/app/**/*.stories.mdx',
    '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
  ],
  addons: ['@storybook/addon-essentials', ...(rootMain.addons || [])],
  async viteFinal(config: any) {
    return mergeConfig(config, {
      plugins: [
        viteTsConfigPaths({
          root: '../../../',
        }),
      ],
    });
  },
};

module.exports = config;
"
`;

exports[`Add addon-essentials to project-level main.js files should add addon-essentials to main.js|ts where possible 3`] = `
"const rootMain = require('../../../.storybook/main');
const { mergeConfig } = require('vite');
const viteTsConfigPaths = require('vite-tsconfig-paths').default;

module.exports = {
  ...rootMain,
  core: { ...rootMain.core, builder: 'webpack5' },
  stories: [
    ...rootMain.stories,
    '../src/app/**/*.stories.mdx',
    '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
  ],
  addons: [
    '@storybook/addon-essentials',
    ...rootMain.addons,
    '@nrwl/react/plugins/storybook',
  ],
  webpackFinal: async (config, { configType }) => {
    // apply any global webpack configs that might have been specified in .storybook/main.js
    if (rootMain.webpackFinal) {
      config = await rootMain.webpackFinal(config, { configType });
    }

    // add your own webpack tweaks if needed

    return config;
  },
  async viteFinal(config, { configType }) {
    return mergeConfig(config, {
      plugins: [
        viteTsConfigPaths({
          root: '../../../',
        }),
      ],
    });
  },
};
"
`;

exports[`Add addon-essentials to project-level main.js files should add addon-essentials to main.js|ts where possible 4`] = `
"// project: react-rollup
const rootMain = require('../../../.storybook/main');
// Use the following syntax to add addons!
rootMain.addons.push('@storybook/addon-essentials', '@storybook/addon-a11y');
rootMain.stories.push(
  ...['../src/lib/**/*.stories.mdx', '../src/lib/**/*.stories.@(js|jsx|ts|tsx)']
);

module.exports = rootMain;

module.exports.core = { ...module.exports.core, builder: 'webpack5' };
"
`;

exports[`Add addon-essentials to project-level main.js files should add addon-essentials to main.js|ts where possible 5`] = `
"// project: react-rollup-3
const rootMain = require('../../../.storybook/main');
// Use the following syntax to add addons!
rootMain.addons.push('@storybook/addon-essentials');
rootMain.stories.push(
  ...['../src/lib/**/*.stories.mdx', '../src/lib/**/*.stories.@(js|jsx|ts|tsx)']
);

module.exports = rootMain;

module.exports.core = { ...module.exports.core, builder: 'webpack5' };
"
`;

exports[`Add addon-essentials to project-level main.js files should add addon-essentials to main.js|ts where possible 6`] = `
"// project: react-rollup-4
const rootMain = require('../../../.storybook/main');
rootMain.addons.push('@storybook/addon-essentials');
rootMain.stories.push(
  ...['../src/lib/**/*.stories.mdx', '../src/lib/**/*.stories.@(js|jsx|ts|tsx)']
);

module.exports = rootMain;

module.exports.core = { ...module.exports.core, builder: 'webpack5' };
"
`;

exports[`Add addon-essentials to project-level main.js files should add addon-essentials to main.js|ts where possible 7`] = `
"const rootMain = require('../../../.storybook/main');
const { mergeConfig } = require('vite');
const viteTsConfigPaths = require('vite-tsconfig-paths').default;

module.exports = {
  ...rootMain,
  core: { ...rootMain.core, builder: 'webpack5' },
  stories: [
    ...rootMain.stories,
    '../src/app/**/*.stories.mdx',
    '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
  ],
  addons: [
    '@storybook/addon-essentials',
    ...rootMain.addons,
    '@nrwl/react/plugins/storybook',
  ],
  webpackFinal: async (config, { configType }) => {
    // apply any global webpack configs that might have been specified in .storybook/main.js
    if (rootMain.webpackFinal) {
      config = await rootMain.webpackFinal(config, { configType });
    }

    // add your own webpack tweaks if needed

    return config;
  },
  async viteFinal(config, { configType }) {
    return mergeConfig(config, {
      plugins: [
        viteTsConfigPaths({
          root: '../../../',
        }),
      ],
    });
  },
};
"
`;

exports[`Add addon-essentials to project-level main.js files should add addon-essentials to main.js|ts where possible 8`] = `
"
        some invalid stuff
        "
`;

exports[`Add addon-essentials to project-level main.js files should remove addon-essentials from root config file 1`] = `
"module.exports = {
  addons: ['@storybook/addon-knobs'],
};
"
`;

exports[`Add addon-essentials to project-level main.js files should remove addon-essentials from root config typescript file 1`] = `
"import type { StorybookConfig } from '@storybook/core-common';

export const rootMain: StorybookConfig = {
  addons: ['@storybook/addon-knobs'],
};
"
`;

exports[`Add addon-essentials to project-level main.js files should remove the root config import and delete root file 1`] = `
"const { mergeConfig } = require('vite');
const viteTsConfigPaths = require('vite-tsconfig-paths').default;

module.exports = {
  core: { builder: 'webpack5' },
  stories: [
    '../src/app/**/*.stories.mdx',
    '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
  ],
  addons: ['@storybook/addon-essentials'],
  async viteFinal(config, { configType }) {
    return mergeConfig(config, {
      plugins: [
        viteTsConfigPaths({
          root: '../../../',
        }),
      ],
    });
  },
};
"
`;

exports[`Add addon-essentials to project-level main.js files should remove the root config import and delete root file 2`] = `
"import type { StorybookConfig } from '@storybook/core-common';
import { mergeConfig } from 'vite';
import viteTsConfigPaths from 'vite-tsconfig-paths';

const config: StorybookConfig = {
  core: { builder: '@storybook/builder-vite' },
  stories: [
    '../src/app/**/*.stories.mdx',
    '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
  ],
  addons: ['@storybook/addon-essentials'],
  async viteFinal(config: any) {
    return mergeConfig(config, {
      plugins: [
        viteTsConfigPaths({
          root: '../../../',
        }),
      ],
    });
  },
};

module.exports = config;
"
`;

exports[`Add addon-essentials to project-level main.js files should remove the root config import and delete root file 3`] = `
"const { mergeConfig } = require('vite');
const viteTsConfigPaths = require('vite-tsconfig-paths').default;

module.exports = {
  core: { builder: 'webpack5' },
  stories: [
    '../src/app/**/*.stories.mdx',
    '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
  ],
  addons: ['@storybook/addon-essentials', '@nrwl/react/plugins/storybook'],
  webpackFinal: async (config, { configType }) => {
    // apply any global webpack configs that might have been specified in .storybook/main.js

    // add your own webpack tweaks if needed

    return config;
  },
  async viteFinal(config, { configType }) {
    return mergeConfig(config, {
      plugins: [
        viteTsConfigPaths({
          root: '../../../',
        }),
      ],
    });
  },
};
"
`;

exports[`Add addon-essentials to project-level main.js files should remove the root config import and delete root file 4`] = `
"// project: react-rollup

// Use the following syntax to add addons!

module.exports = rootMain;

module.exports.core = { ...module.exports.core, builder: 'webpack5' };
module.exports.addons = [
  '@storybook/addon-essentials',
  '@storybook/addon-a11y',
];
module.exports.stories = [
  '../src/lib/**/*.stories.mdx',
  '../src/lib/**/*.stories.@(js|jsx|ts|tsx)',
];
"
`;

exports[`Add addon-essentials to project-level main.js files should remove the root config import and delete root file 5`] = `
"// project: react-rollup-3

// Use the following syntax to add addons!

module.exports = rootMain;

module.exports.core = { ...module.exports.core, builder: 'webpack5' };
module.exports.addons = ['@storybook/addon-essentials'];
module.exports.stories = [
  '../src/lib/**/*.stories.mdx',
  '../src/lib/**/*.stories.@(js|jsx|ts|tsx)',
];
"
`;

exports[`Add addon-essentials to project-level main.js files should remove the root config import and delete root file 6`] = `
"// project: react-rollup-4

module.exports = rootMain;

module.exports.core = { ...module.exports.core, builder: 'webpack5' };
module.exports.addons = ['@storybook/addon-essentials'];
module.exports.stories = [
  '../src/lib/**/*.stories.mdx',
  '../src/lib/**/*.stories.@(js|jsx|ts|tsx)',
];
"
`;

exports[`Add addon-essentials to project-level main.js files should remove the root config import and delete root file 7`] = `
"const { mergeConfig } = require('vite');
const viteTsConfigPaths = require('vite-tsconfig-paths').default;

module.exports = {
  core: { builder: 'webpack5' },
  stories: [
    '../src/app/**/*.stories.mdx',
    '../src/app/**/*.stories.@(js|jsx|ts|tsx)',
  ],
  addons: ['@storybook/addon-essentials', '@nrwl/react/plugins/storybook'],
  webpackFinal: async (config, { configType }) => {
    // apply any global webpack configs that might have been specified in .storybook/main.js

    // add your own webpack tweaks if needed

    return config;
  },
  async viteFinal(config, { configType }) {
    return mergeConfig(config, {
      plugins: [
        viteTsConfigPaths({
          root: '../../../',
        }),
      ],
    });
  },
};
"
`;

exports[`Add addon-essentials to project-level main.js files should remove the root config import and delete root file 8`] = `
"
        some invalid stuff
        "
`;
